src/status_im/contexts/wallet/swap/events.cljs

```diff
@@ -19,16 +19,17 @@
          test-networks-enabled? (get-in db [:profile/profile :test-networks-enabled?])
          view-id                (:view-id db)
          root-screen?           (or (= view-id :wallet-stack) (nil? view-id))
-         account                (or from-account (swap-utils/wallet-account wallet))
+         available-accounts     (utils/get-accounts-with-token-balance (:accounts wallet)
+                                                                       (:asset-to-pay data))
+         account                (or from-account (swap-utils/wallet-account wallet available-accounts))
```

Comment:
As far as I understood, the logic for selecting account it is:
`from-account` OR `current-viewing-account` OR `(first available-accounts)`

But in current implementation this logic split in two parts. One `OR` happens here, in the event code and another `OR` inside `swap-utils/wallet-acocunt`.
I'd suggest to make it more obvious. 
We can move all the logic inside `swap-utils/wallet-account` and also rename it to something more obvious, like `swap-utils/swap-from-account`. 
Another approach is to simplify `swap-utils/wallet-account` to only contain `current-viewing-account` (and rename it appropriately and make all `OR`s inside the event code:
```clojure
[account (or from-account 
             (swap-utils/current-viewing-account)
             (first available-accounts))]
```

This will make the logic easier to comprehend by code reader. Wdyt?

Reply:
@vkjr make sense, pushed changes